apps/backend/src/services/responseServices.ts

```diff
@@ -0,0 +1,195 @@
+import { Response } from '@prisma/client';
+import { ResponseData, UpdateResponseData } from '../types';
+import prismaClient from '../../prismaClient';
+import middleware from '../utils/middleware';
+import requestServices from './requestServices';
+import neighborhoodServices from './neighborhoodServices';
+
+/**
+ * performs narrowing on object `obj`
+ * returns `true` if `obj` contains required properties for
+ * `ResponseData` type
+ * @param obj - request.body
+ * @returns - type predicate (boolean)
+ */
+const isResponseData = (obj: object): obj is ResponseData => 'content' in obj
```

Comment:
Can you explain what this line of code does? I find it very interesting.

Reply:
+1